File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        seen = set()
        seen.add(head)
        trav = head
        while trav:
            trav = trav.next
            if trav and trav in seen:
                return True
            elif trav:
                seen.add(trav)
        return False

File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev = nullptr;
        ListNode* next;
        for (ListNode* trav=head; trav != nullptr;) {
            cout << trav->val;
            next = trav->next;
            trav->next = prev;
            prev = trav; 
            trav = next;
            //if (prev) { cout << prev->val << endl ; }
            //if (trav && prev) { cout << ' ' << prev->val << ' ' << trav->val << endl; }
        }
        return prev;
    }
};
